docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+- Develop using the Python and UI SDKs
+- Exchange data between UI and Python SDK processes; during both `UPDATE` and `RUN` modes
+- Use metaprogramming, Python SDK, and the UI SDK to create an editable configuration and dynamic behavior between or during workflow runs
+- Package and use 3rd party libraries from `npm` and `pypi` in a tool
+- Develop "ready to use" Keras models from scratch, including models for deploying into production workflows.
+
+- [1. Create a Basic Text Classifier Neural Network with Tensorflow](#1-create-a-basic-text-classifier-nueral-network-with-tensorflow)
+  - [1.1. Prerequisites](#11-prerequisites)
+  - [1.2. Overview](#12-overview)
+    - [1.2.1. Data](#121-data)
+    - [1.2.2. Preview](#122-preview)
+    - [1.2.3. TRAIN](#123-train)
+    - [1.2.4. Predict](#124-predict)
+    - [1.2.5. Create a Plugin](#125-create-a-plugin)
+      - [1.2.5.1. UI/TextClassifier/src/index.tsx](#1251-uitextclassifiersrcindextsx)
+      - [1.2.5.2. backend/ayx\_plugins/text\_classifier.py](#1252-backendayx_pluginstext_classifierpy)
+  - [1.3. Backend: A Naive Tensorflow Text Classifier](#13-backend-a-naive-tensorflow-text-classifier)
+    - [1.3.1. Data Mode](#131-data-mode)
+      - [Supplemental: Error Handling in the SDK](#supplemental-error-handling-in-the-sdk)
+      - [Adding a custom method: `setup_data()`](#adding-a-custom-method-setup_data)
+    - [](#)
+
+
+## 1.1. Prerequisites
+
+While we do our best to provide enough information here for anyone with general scripting experience to follow along, the recommended prereqs are provided below for general debugging and troubleshooting knowledge outside of the scope possible here:
+
+* Tensorflow
+  * Keras
+  * Datasets
+* React and Typescript
+* General understanding or knowledge of Neural Networks and AI/ML algorithms.
+
+## 1.2. Overview
+
+The tool we create will allow a user to generate a Text Classifier NN(Neural Network) from scratch, train it using User-defined datasets, evaluate training, and then predict with a "production" version of said model.
+As mentioned, we'll aim to recreate Tensorflow's Official tutorial model as a workflow tool. If you run into TensorFlow related issues, we recommend you review and cross-check the code provided in the tutorial.
+
+With that in mind, we create a "naive" version of our text classifier tool.
+That is, a tool without any UI to ensure critical functions are running as intended.
+To achieve this, we also mock data and relevant calls to represent what will come from the UI SDK to the Python backend and what the backend will send to the UI SDK via `save_config()` (link to docs here). 
+Next, we build our front end using the UI SDK using its provided `dev-harness`.
+Once the back and front ends are complete, we then bundle and deploy using the CLI to do end-to-end testing.
+Finally, we package our dependencies and source for deployment using `create-yxi`.
+
+The tool will have 4 "modes": `DATA`, `PREVIEW`, `TRAIN`, and `PREDICT`. 
```

Comment:
TW review:

The tool has 4 "modes": `DATA`, `PREVIEW`, `TRAIN`, and `PREDICT`.